app/src/main/java/com/sdp/swiftwallet/CryptoGraphActivity.java

```diff
@@ -50,27 +63,98 @@ protected void onCreate(Bundle savedInstanceState) {
         // Get the Intent that started this activity and extract the currency
         Intent intent = getIntent();
         currency = (Currency) intent.getSerializableExtra("currency");
-        //rateSymbol = currency.getSymbol()+"USDT";
-        rateSymbol = "ETH"+"USDT";
+        if(currency == null) currency = new Currency("Ethereum", "ETH", 2000);
+        else rateSymbol = currency.getSymbol()+"USDT";
         interval = "1h";
 
         // Get the data from Binance API
         mIdlingResource = new CountingIdlingResource("CryptoValue Calls");
+        setIntervalsSpinner();
         mIdlingResource.increment();
-        getData();
+        candleStickChart = getData();
+        mIdlingResource.decrement();
+
+        /*
+        System.out.println("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
+        System.out.println(openTimes.get(3));
+
+
+        //Create CandleStickChart
+        CandleStickChart candleStickChart = createCandleStickChart();
+        //get data as CandleData
+        CandleData candleData = createDataSetForCandleGraph(openTimes, openValues, highValues, lowValues, closeValues, volumeValues, closeTimes);
+        //setData
+        candleStickChart.setData(candleData);
+        candleStickChart.invalidate();
+        */
 
 
         // Capture the layout's TextView and set the string as its text
         TextView textView = findViewById(R.id.idCurrencyToShowName);
         //textView.setText(closeTimes.size());
-        textView.setText("Ethereum");
+        textView.setText(currency.getName());
 
         //textView.setText("LAST CLOSING TIME : "+closeTimes.get(closeTimes.size()-1) + " LAST CLOSING VALUE : "+closeValues.get(closeValues.size()-1));
     }
 
-    private void getData(){
-        String url = "https://api.binance.com/api/v1/klines?symbol=ETHUSDT&interval=1h";
+    private CandleStickChart createCandleStickChart(){
+        CandleStickChart candleStickChart = findViewById(R.id.candle_stick_chart);
+        candleStickChart.setHighlightPerDragEnabled(true);
+
+        //CHECK FALSE
+        candleStickChart.setDrawBorders(true);
+
+        candleStickChart.setBorderColor(Color.LTGRAY);
+
+        YAxis yAxis = candleStickChart.getAxisLeft();
+        YAxis rightAxis = candleStickChart.getAxisRight();
+        yAxis.setDrawGridLines(false);
+        rightAxis.setDrawGridLines(false);
+        candleStickChart.requestDisallowInterceptTouchEvent(true);
+
+        XAxis xAxis = candleStickChart.getXAxis();
+        xAxis.setDrawGridLines(false);
+        xAxis.setDrawLabels(true);
+        rightAxis.setTextColor(Color.BLACK);
+        yAxis.setDrawLabels(false);
+        xAxis.setGranularity(1f);
+        xAxis.setGranularityEnabled(true);
+        xAxis.setAvoidFirstLastClipping(true);
```

Comment:
This is kind of cumbersome, isn't there a way to configure the axis in another way ?